src/main/kotlin/org/sert2521/infiniterecharge2020/utils/Utils.kt

```diff
@@ -0,0 +1,26 @@
+package org.sert2521.infiniterecharge2020.utils
+
+import kotlin.math.sign
+import org.sert2521.sertain.control.PidfConfig
+
+class PidfController2(config: PidfConfig, val dt: Double) {
+    private val kp = config.kp ?: 0.0
+    private val ki = config.ki ?: 0.0
+    private val kd = config.kd ?: 0.0
+    private val kf = config.kf ?: 0.0
+
+    private var integral = 0.0
+    private var lastError = 0.0
+
+    fun next(setPoint: Double, actual: Double): Double {
+        val error = setPoint - actual
+        integral += error * dt
+        val derivative = (error - lastError) / dt
+        lastError = error
+        return (kp * error) + (ki * integral) + (kd * derivative) + (kf * sign(error))
+    }
+}
+
+// So ktlint can be happy
+fun placeHolder() {
```

Comment:
I don't know ktlint that well. Why do we need a function at the end of the file?

Reply:
I assume this is because ktlint complains that the name of the file is different from the name of the class, but I think you should be able to disable that rule as it's a bit silly in Kotlin (more of a Java requirement).

Reply:
If someone wants to turn off a specific rule, here is a link to the ktlint plugin docs:
https://github.com/jlleitschuh/ktlint-gradle#configuration